packages/loot-core/src/server/accounts/sync.ts

```diff
@@ -61,6 +86,39 @@ async function updateAccountBalance(id, balance) {
   ]);
 }
 
+async function getAccountSyncStartDate(id) {
+  // there is no way to get the current date in sql.js, the workaround is to
+  // calculate the date in js and pass it into the query
+  const currentDate = new Date().toISOString().slice(0, 19).replace('T', ' ');
+  const latestTransaction = await db.first(
+    'SELECT * FROM v_transactions WHERE account = ? AND date <= ? ORDER BY date DESC LIMIT 1',
+    [id, currentDate],
+  );
+
+  if (!latestTransaction) return null;
+
+  const startingTransaction = await db.first(
+    'SELECT date FROM v_transactions WHERE account = ? ORDER BY date ASC LIMIT 1',
```

Comment:
Would it be possible to use aql here?

Reply:
Done, cleaned up some other problems around this function too